### PR TITLE
repo/checkout: Finish conversion to new code style

### PR DESCRIPTION
**Pull request by cgwalters**
I plan to make some future changes here, and wanted to do this
first.

Random side note; how about converting the do/while loops for `EINTR` to
`TEMP_FAILURE_RETRY()`? We're very inconsistent about that...

### CODE REVIEW

File: src/libostree/ostree-repo-checkout.c [hunk 1]
```diff
@@ -420,21 +378,12 @@ checkout_one_file_at (OstreeRepo                        *repo,
...
   char loose_path_buf[_OSTREE_LOOSE_PATH_MAX];
-  g_autoptr(GInputStream) input = NULL;
-  g_autoptr(GVariant) xattrs = NULL;
-  gboolean is_whiteout;
-
-  is_symlink = g_file_info_get_file_type (source_info) == G_FILE_TYPE_SYMBOLIC_LINK;
-  checksum = ostree_repo_file_get_checksum ((OstreeRepoFile*)source);
-
-  is_whiteout = !is_symlink && options->process_whiteouts &&
+  const gboolean is_symlink = g_file_info_get_file_type (source_info) == G_FILE_TYPE_SYMBOLIC_LINK;
```
Comment (jlebon) [inline src/libostree/ostree-repo-checkout.c]: Can we wrap in parentheses?

File: src/libostree/ostree-repo-checkout.c [hunk 2]
```diff
@@ -420,21 +378,12 @@ checkout_one_file_at (OstreeRepo                        *repo,
...
+  const char *checksum = ostree_repo_file_get_checksum ((OstreeRepoFile*)source);
+  const gboolean is_whiteout = !is_symlink && options->process_whiteouts &&
```
Comment (jlebon) [inline src/libostree/ostree-repo-checkout.c]: And here also!

File: src/libostree/ostree-repo-checkout.c [hunk 3]
```diff
@@ -516,24 +458,21 @@ checkout_one_file_at (OstreeRepo                        *repo,
...
-                goto out;
+                return FALSE;
 
               if (hardlink_res == HARDLINK_RESULT_LINKED && options->devino_to_csum_cache)
                 {
                   struct stat stbuf;
                   OstreeDevIno *key;
-                  
+
                   if (TEMP_FAILURE_RETRY (fstatat (destination_dfd, destination_name, &stbuf, AT_SYMLINK_NOFOLLOW)) != 0)
```
Comment (jlebon) [inline src/libostree/ostree-repo-checkout.c]: Re. your comment, I think my vote is for `TEMP_FAILURE_RETRY`. I find it easier to read (plus, it's even less boilerplate!).

File: src/libostree/ostree-repo-checkout.c [hunk 4]
```diff
@@ -712,65 +634,59 @@ checkout_tree_at (OstreeRepo                        *self,
...
   if (g_file_info_get_file_type (source_info) != G_FILE_TYPE_DIRECTORY)
     {
-      ret = checkout_one_file_at (self, options,
-                                  (GFile *) source,
-                                  source_info,
-                                  destination_dfd,
-                                  g_file_info_get_name (source_info),
-                                  cancellable, error);
-      goto out;
+      if (!checkout_one_file_at (self, options,
```
Comment (jlebon) [inline src/libostree/ostree-repo-checkout.c]: We can just `return` the value directly, right?

File: src/libostree/ostree-repo-checkout.c (hunk 2 above)
Comment (cgwalters) [inline src/libostree/ostree-repo-checkout.c]: Note this was just a moved hunk, not net new code.  But sure.  Fixups :arrow_down:

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 868e374

**Pull request closed without merging**